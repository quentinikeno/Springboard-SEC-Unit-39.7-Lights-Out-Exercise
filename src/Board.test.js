import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Board from "./Board";

describe("<Board /> rendering", () => {
	test("initial board should render without crashing", () => {
		render(<Board />);
	});

	test("Should match snapshot when all cells are lit", () => {
		const { asFragment } = render(<Board chanceLightStartsOn={1} />);
		expect(asFragment()).toMatchSnapshot();
	});

	test("Should match snapshot when all cells are not lit", () => {
		const { asFragment } = render(<Board chanceLightStartsOn={0} />);
		expect(asFragment()).toMatchSnapshot();
	});
});

describe("clicking cells", () => {
	test("Lights should toggle correctly", () => {
		const { getAllByRole } = render(
			<Board nrows={3} ncols={3} chanceLightStartsOn={1} />
		);
		const cells = getAllByRole("button");

		cells.forEach((cell) => {
			expect(cell).toHaveClass("Cell-lit");
		});

		fireEvent.click(cells[4]);

		let litIndices = [0, 2, 6, 8];
		cells.forEach((cell, index) => {
			if (litIndices.includes(index)) {
				expect(cell).toHaveClass("Cell-lit");
			} else {
				expect(cell).not.toHaveClass("Cell-lit");
			}
		});
	});

	test("that the win message appears on game win", () => {
		const { queryByText, getAllByRole } = render(
			<Board nrows={1} ncols={3} chanceLightStartsOn={1} />
		);

		expect(queryByText("You've won!")).not.toBeInTheDocument();

		const cells = getAllByRole("button");
		fireEvent.click(cells[1]);
		expect(queryByText("You've won!")).toBeInTheDocument();
	});
});
