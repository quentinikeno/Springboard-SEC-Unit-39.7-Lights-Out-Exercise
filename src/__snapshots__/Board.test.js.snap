// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> rendering Should match snapshot when all cells are lit 1`] = `
<DocumentFragment>
  <table
    class="Board"
  >
    <tbody>
      <tr>
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
      </tr>
      <tr>
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
        <td
          class="Cell Cell-lit"
          role="button"
        />
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`<Board /> rendering Should match snapshot when all cells are not lit 1`] = `
<DocumentFragment>
  <h2>
    You've won!
  </h2>
</DocumentFragment>
`;
